### PR TITLE
Builds the `WinterKill` component

### LINKED ISSUE #311: Build a `WinterKill` component
Create a component for indicating if a Cover Crop Seeding will be winter killed.

This component should appear approximately as follows:
image

Notes:
- The date field should only appear if the box is checked.
- The date field should default to Jan 1 of the next year.
- This component does not need to have a `required` prop.
- This component should have props for the checkbox state and the date.
- This component should emit update events when the checkbox state or the date change.

Building this component should also include:
- An example page in `farm_fd2_examples` for the component.
- A set of component tests for the component.

### PR DESCRIPTION
**Pull request by Shahir-47**
**Pull Request Description**

Creates a component for indicating if a Cover Crop Seeding will be winter killed. Additionally, it creates an example page in `farm_fd2_examples` for the component and a set of component tests for the component.

Closes #311 
---

**Licensing Certification**

FarmData2 is a [Free Cultural Work](https://freedomdefined.org/Definition) and all accepted contributions are licensed as described in the LICENSE.md file. This requires that the contributor holds the rights to do so. By submitting this pull request **I certify that I satisfy the terms of the [Developer Certificate of Origin](https://developercertificate.org/)** for its contents.

### CODE REVIEW

**Comment by braughtg** [comment]
Flip this to a full PR when you are ready for me to have a look at it.

**Comment by Shahir-47** [comment]
> Flip this to a full PR when you are ready for me to have a look at it.

Ready for review!

File: modules/farm_fd2_examples/src/entrypoints/winter_kill/App.vue [hunk 1]
```diff
@@ -0,0 +1,203 @@
+<template>
+  <div>
+    <h3>WinterKill Example</h3>
+    <p>
+      The WinterKill component allows the user to specify if a crop seeding will
```
Comment (braughtg) [inline modules/farm_fd2_examples/src/entrypoints/winter_kill/App.vue]: Change to "... if a crop will be winter killed."  (drop "seeding")

File: modules/farm_fd2_examples/src/module/farm_fd2_examples.links.menu.yml [hunk 2]
```diff
@@ -58,3 +58,9 @@ farm.fd2_examples_example_entry_point:
   parent: farm.fd2_examples_main
   route_name: farm.fd2_examples_example_entry_point.content
   weight: 100
+farm.fd2_examples_winter_kill:
+  title: 'WinterKill'
+  description: 'Example of the use of the WinterKIll Component.'
```
Comment (braughtg) [inline modules/farm_fd2_examples/src/module/farm_fd2_examples.links.menu.yml]: `I` should be `i` in `WinterKill` here.

File: components/WinterKill/WinterKill.vue [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+ * ```
+ *
+ * ## `data-cy` Attributes
+ *
+ * Attribute Name                      | Description
+ * ------------------------------------| -----------
+ * winter-kill-group                   | The `BFormGroup` component containing the winter kill checkbox and date input fields.
+ * winter-kill-label                   | The `span` component containing the "Winter kill:" label.
+ * winter-kill-checkbox                | The `BFormCheckbox` component used to indicate if the crop will be winter killed.
+ * winter-kill-date-group              | The `div` component containing the date input fields when the checkbox is selected.
```
Comment (braughtg) [inline components/WinterKill/WinterKill.vue]: I'm not quite sure what this div contains from this comment.  Is it just the one date field? If so use `field` here not `fields`.  Or is there more in here than the date field? If so, try to be more specific.

**Review by braughtg — CHANGES_REQUESTED** [review]
The functionality here looks pretty good. A few small things in the code and:
- Can we reduce the space between the label and the checkbox?
- Can we reduce the space between the checkbox and the date input?

Hopefully enough that the calendar component appears reliably in mobile mode for iPhone SE.

With those things touched up, it will just need to have the tests implemented.

**Comment by Shahir-47** [comment]
> The functionality here looks pretty good. A few small things in the code and:
> 
> * Can we reduce the space between the label and the checkbox?
> * Can we reduce the space between the checkbox and the date input?
> 
> Hopefully enough that the calendar component appears reliably in mobile mode for iPhone SE.
> 
> With those things touched up, it will just need to have the tests implemented.

I can't change the space between the label and the checkbox as that is the default padding that comes with the label and the input column to the right of the label. If you remember last week, we couldn't change `<div class="col">` and `<label class="col-auto col-form-label col-form-label text-end" id="__BVID__474873___BV__BV_label___" for="winter-kill-checkbox">` because that is generated by `<BFormGroup>` and it overwrites whatever style  applied to `<div class="col">` and `<label class="col-auto col-form-label col-form-label text-end" id="__BVID__474873___BV__BV_label___" for="winter-kill-checkbox">`

 You can see the padding for the input col here:

image

 You can see the padding for the label here:

image


However, I can change the spacing between the checkbox and the date input.

**Comment by braughtg** [comment]
However, I can change the spacing between the checkbox and the date input.

Sounds good. Try to shrink that down and see how we do with the calendar that pops up.

**Comment by Shahir-47** [comment]
> However, I can change the spacing between the checkbox and the date input. Sounds good. Try to shrink that down and see how we do with the calendar that pops up.

It does show on the iPhone SE if I double-click. Here's the mobile view with the spacing reduced: 

image

Desktop view: 

image

**Comment by Shahir-47** [comment]
Another related issue I encountered: When I tried to commit my changes the first time, winter_kill.exists.e2e.cy.js  always failed. However, when I run winter_kill.exists.e2e.cy.js and then commit it, the test passes, and the changes are committed with no issues.

**Comment by braughtg** [comment]
> It does show on the iPhone SE if I double-click. 

It was showing for me about 50% of the time when I double clicked.  This may be resolved when actually running on mobile.  It sounds like the Bootstrap Vue Next uses the native date component when running on the device. When I get back we can try to launch this on a site we can access via mobile and check it out.

> Here's the mobile view with the spacing reduced:

Looks better! Nice work.

**Comment by braughtg** [comment]
> Another related issue I encountered: When I tried to commit my changes the first time, winter_kill.exists.e2e.cy.js always failed. However, when I run winter_kill.exists.e2e.cy.js and then commit it, the test passes, and the changes are committed with no issues.

Strange behavior.  I'm not sure what was going on here without seeing the actual runs or some output showing how / why the test failed.  If it persists, please open an issue for it with as much detail as you can gather about the test failure.

File: components/WinterKill/WinterKill.vue
Comment (braughtg) [inline components/WinterKill/WinterKill.vue:1]: One of the `emits` for each event should be commented to describe the event and its payload (if it has one).  For example, this is what the `EquipmentSelector` does for the `valid` event:
```Javascript
    isValid() {
      /**
       * The validity of the selected equipment changed.
       * @property {boolean} event whether the selected equipment is valid or not.
       */
      this.$emit('valid', this.isValid);
    },
```

If an event is emitted from multiple places, only one needs to have the JSDoc comment.

**Review by braughtg — CHANGES_REQUESTED** [review]
Looking good! Just one comment on `WinterKill.vue` regarding JSDoc for events.

**Pull request merged**